src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/edit_index_pattern.js

```diff
@@ -140,4 +142,45 @@ uiModules.get('apps/management')
       $scope.indexPattern.timeFieldName = field.name;
       return $scope.indexPattern.save();
     };
+
+    $scope.tryToRenderScriptedFieldsTable = function () {
+      if ($state.tab === 'scriptedFields') {
+        $scope.$$postDigest($scope.renderScriptedFieldsTable);
+      } else {
+        destroyScriptedFieldsTable();
+      }
+    };
+
+    $scope.renderScriptedFieldsTable = function () {
```

Comment:
I think it makes sense to redefine this as a local function instead of a scope method (unless it needs to be on scope, but I don't think it does), and to move the logic from `scripted_fields_table/index.js` into here instead. So it would be:

```jsx
function renderScriptedFieldsTable() {
  const node = document.getElementById(DOM_ELEMENT_ID);
  if (!node) {
    return;
  }

  render(
    <ScriptedFieldsTable
      indexPattern={$scope.indexPattern}
      fieldFilter={$scope.fieldFilter}
      scriptedFieldLanguageFilter={$scope.scriptedFieldLanguageFilter}
      helpers={{
        redirectToRoute: (obj, route) => {
          $scope.kbnUrl.redirectToRoute(obj, route);
          $scope.$apply();
        },
        getRouteHref: (obj, route) => $scope.kbnUrl.getRouteHref(obj, route),
      }}
    />,
    node,
  );
}
```

This makes sense to me since it mirrors how this will be structured once we migrate to React. The owner of the `<ScriptedFieldsTable />` instance will be responsible for providing the props it needs. I think it also simplifies the code a bit.

Reply:
Sounds good!